### PR TITLE
Kotlin

### PR DESCRIPTION
**Pull request by obj-obj**
Please actually consider this, I think Kotlin is a much better language than java

Basically, summing up my argument:

- Kotlin has little to no learning curve for ppl coming from java, because I can port java code to kotlin after under 30 mins on the docs, and because I showed it to multiple people in the programming subteam who've never seen kotlin before, and they understood the code fine without even looking at the docs

- Kotlin has non-nullable types, which can and will prevent some runtime errors

- Kotlin is more concise than java, which might have a positive impact on development times

- I have handled porting all existing code to kotlin - unless there's changes people haven't pushed yet

- Kotlin has full interop with java - and since it compiles to a .jar as well, simulations/deploy/etc will work fine

### CODE REVIEW

**Comment by obj-obj** [comment]
I brought the code to a few people who have never seen kotlin before and they understood it fine, so perhaps no overhead

I've also been on the kotlin docs for a total of like 30 minutes

We can probably save more time than that writing more concise code

File: build.gradle [hunk 1]
```diff
@@ -82,7 +82,6 @@ dependencies {
     testImplementation 'junit:junit:4.12'
 
     errorprone group: 'com.google.errorprone', name: 'error_prone_core', version: '2.10.0'
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
```
Comment (obj-obj) [inline build.gradle:85]: I removed this because the kotlin plugin already automatically imports a relevant version

File: build.gradle [hunk 2]
```diff
@@ -108,11 +107,11 @@ wpi.java.configureExecutableTasks(jar)
 wpi.java.configureTestTasks(test)
 compileKotlin {
     kotlinOptions {
-        jvmTarget = "1.8"
+        jvmTarget = "11"
```
Comment (obj-obj) [inline build.gradle:110]: Change kotlin and java JVM targets to be the same

**Comment by obj-obj** [comment]
I will also handle all the porting work

**Comment by obj-obj** [comment]
Basically, summing up my argument:

- Kotlin has little to no learning curve for ppl coming from java, because I can port java code to kotlin after under 30 mins on the docs, and because I showed it to multiple people in the programming subteam who've never seen kotlin before, and they understood the code fine without even looking at the docs

- Kotlin has non-nullable types, which can and will prevent some runtime errors

- Kotlin is more concise than java, which might have a positive impact on development times

- I can handle the work of porting existing code to Kotlin (I've already done most of the work, anyway)

- Kotlin has full interop with java

**Comment by obj-obj** [comment]
We have go from mentors if all subteams agree
pog

**Comment by obj-obj** [comment]
Another mentor says no :(

idk whoever reviews PRs can decide

**Comment by obj-obj** [comment]
I think this PR is ready, the only files that aren't Kotlin are the template files from WPILIB

**Comment by obj-obj** [comment]
There are no java files left in the repo now

**Comment by 1198159** [comment]
no

**Comment by obj-obj** [comment]
> no

yes
java bad kotlin good

**Review by jamesdooley4 — CHANGES_REQUESTED** [review]
Let's chat more as a team about this on Friday.

**Comment by obj-obj** [comment]
> Let's chat more as a team about this on Friday.

Yeah I was planning to do that, most ppl seemed against it on Monday but I've been maintaining it during the week just in case

**Comment by obj-obj** [comment]
Team discussion -> whole repo shouldn't be in kotlin (not really surprising, but I might as well try)

**Pull request closed without merging**